### PR TITLE
feat: optimizations to get more experience per hour ( 84k XP/H )

### PR DESCRIPTION
**Pull request by matheussampaio**
**THIS CODE ISN'T INTENDED TO BE MERGED.**

It's here because some people on Slack asked me to, so they can test and discuss.

# Short Description:
A bunch of optimizations to improve the experience per hour. 

# Main changes:
- Add a loop to Spin Fort worker, now we spin every ready and in range fort per tick.
- Add a loop in CatchLuredPokemon, same as above.
- Add a loop in CatchVisiblePokemon, same as above.
- Spin fort even with full inventory
- Move to fort even with full inventory
- Also filter forts with `cooldown_complete_timestamp_ms` (to spinning and moving to)
- Remove a bunch of logs (it's hard to see what happens)
- Change some logs (to make it easy to see the experience)
- Add PR #2252 (thanks @Moonlight-Angel) and PR #2440 (thanks @lucasfevi)
- Add minimum 5 seconds to each tick (avoid softbans)
- Fixed logs with colors (that extra space freaked me out)

# NOTE:
- In this commit, EvolveAll worker, take 25+ seconds to complete. So DON'T use it.
      Just evolve after catching pokemons.
- I based this commit on e58bf3d57c1c74d7847fdfa55b08410f48917bd0, so won't merge with the latest dev.
- Don't forget to execute `virtualenv .` and `pip install -r requirements.txt`

# **BREAKING CHANGES:**
- Add tick_count and delayed some workers (RecycleItem, TransferPokemon).
```diff
- def work(self):
+ def work(self, *args, **kwargs):
```
- Could be more.

## Recommended `config.json`:
❗   **CHANGE RELEASE RULES AND RECYCLE ITEMS RULES** ❗ 

```
{
    "auth_service": "google",
    "username": "xxx",
    "password": "xxx",
    "location": "40.774626,-73.972706",
    "gmapkey": "xxx",
    "tasks": [
        {
            "type": "HandleSoftBan"
        },
        {
            "type": "UpdateTitleStats",
            "config": {
                "min_interval": 60,
                "stats": ["uptime", "km_walked", "level_stats", "xp_earned", "xp_per_minute", "stops_visited", "pokemon_caught", "pokemon_evolved"]
            }
        },
        {
            "type": "RecycleItems",
            "config": {
                "item_filter": {
                    "Pokeball": {
                        "keep": 80
                    },
                    "Greatball": {
                        "keep": 80
                    },
                    "Ultraball": {
                        "keep": 80
                    },
                    "Masterball": {
                        "keep": 80
                    },
                    "Razz Berry": {
                        "keep": 50
                    },
                    "Potion": {
                        "keep": 0
                    },
                    "Super Potion": {
                        "keep": 0
                    },
                    "Hyper Potion": {
                        "keep": 0
                    },
                    "Max Potion": {
                        "keep": 0
                    },
                    "Revive": {
                        "keep": 0
                    },
                    "Max Revive": {
                        "keep": 0
                    }
                }
            }
        },
        {
            "type": "CollectLevelUpReward"
        },
        {
            "type": "IncubateEggs",
            "config": {
                "longer_eggs_first": true
            }
        },
        {
            "type": "CatchVisiblePokemon"
        },
        {
            "type": "CatchLuredPokemon"
        },
        {
            "type": "TransferPokemon"
        },
        {
            "type": "SpinFort"
        },
        {
            "type": "MoveToFort",
            "config": {
                "lure_attraction": true,
                "lure_max_distance": 2000
            }
        }
    ],
    "map_object_cache_time": 5,
    "max_steps": 150,
    "forts": {
        "avoid_circles": true,
        "max_circle_size": 100
    },
    "websocket_server": true,
    "walk": 80,
    "action_wait_min": 0,
    "action_wait_max": 1,
    "debug": false,
    "test": false,
    "health_record": false,
    "location_cache": false,
    "distance_unit": "km",
    "reconnecting_timeout": 15,
    "evolve_captured": "Pidgey,Rattata,Zubat,Weedle,Drowzee,Spearow,Venonat,Caterpie,Krabby,Paras,Jigglypuff,Sandshrew,Vulpix,Meowth,Psyduck,Mankey,Tentacool,Ponyta,Slowpoke,Magnemite,Doduo,Seel,Voltorb,Horsea,Goldeen,Staryu,Ekans",
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "catch": {
        "any": {
            "catch_above_cp": 0,
            "catch_above_iv": 0,
            "logic": "or"
        }
    },
    "release": {
        
    },
    "vips" : {
        "any": {
            "catch_above_cp": 1200,
            "catch_above_iv": 0.9,
            "logic": "or"
        },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {}
    }
}
```

## How to:

**Install:**
```shell
$ git clone https://github.com/PokemonGoF/PokemonGo-Bot
$ cd PokemonGo-Bot
$ git fetch origin pull/2491/head:more-is-better
$ git checkout more-is-better
$ virtualenv .
$ source bin/activate
$ pip install -r requirements.txt
````

**Start:**
```shell
$ ./pokecli.py -cf configs/config.json
```

## Stats:

@matheussampaio
```
[09:15:26] Ran for 6:25:14
[09:15:26] Total XP Earned: 520285  Average: 81033.48/h
[09:15:26] Travelled 19.71km
[09:15:26] Visited 2051 stops
[09:15:26] Encountered 1508 pokemon, 1431 caught, 0 released, 203 evolved, 1 never seen before
[09:15:26] Threw 1875 pokeballs
[09:15:26] Earned 156165 Stardust
[09:15:26]
[09:15:26] Highest CP Pokemon: Golduck [CP: 1696] [IV: 6/1/15] Potential: 0.49
[09:15:26] Most Perfect Pokemon: Exeggcute [CP: 621] [IV: 14/15/14] Potential: 0.96
```

@lucasfevi:
```
[15:21:14] Ran for 7:27:34
[15:21:14] Total XP Earned: 645165  Average: 86488.98/h
[15:21:14] Travelled 15.90km
[15:21:14] Visited 2721 stops
[15:21:14] Encountered 1776 pokemon, 1704 caught, 0 released, 261 evolved, 8 never seen before
[15:21:14] Threw 2157 pokeballs
[15:21:14] Earned 174807 Stardust
[15:21:14]
[15:21:14] Highest CP Pokemon: Pinsir [CP: 1724] [IV: 14/12/13] Potential: 0.87
[15:21:14] Most Perfect Pokemon: Poliwag [CP: 626] [IV: 13/15/15] Potential: 0.96
```

@w0rd3r (with lucky egg):
```
Uptime : 0:58:58 | 3.29km walked | Level 26 (292,835 / 390,000, 75%) | +216,080 XP | 3,662 XP/m | Visited 4.36 /m | Caught 3.85 /m | Evolved 1.61 /m
```

@theVdude:
```
Uptime : 0:31:21 | 2.36km walked | Level 29 (562,835 / 650,000, 86%) | +126,160 XP | 4,069 XP/m | Visited 4.68 /m | Caught 2.97 /m | Evolved 2.39 /m
```

### CODE REVIEW

**Comment by solderzzc** [comment]
We have to wait for https://github.com/PokemonGoF/PokemonGo-Bot/pull/2173 commit in, that's important changes and douglas keep doing the conflict resolving in 3 days.

**Comment by lucasfevi** [comment]
I've been testing this along with him and it's working pretty well.

```
[15:21:14] Ran for 7:27:34
[15:21:14] Total XP Earned: 645165  Average: 86488.98/h
[15:21:14] Travelled 15.90km
[15:21:14] Visited 2721 stops
[15:21:14] Encountered 1776 pokemon, 1704 caught, 0 released, 261 evolved, 8 never seen before
[15:21:14] Threw 2157 pokeballs
[15:21:14] Earned 174807 Stardust
[15:21:14]
[15:21:14] Highest CP Pokemon: Pinsir [CP: 1724] [IV: 14/12/13] Potential: 0.87
[15:21:14] Most Perfect Pokemon: Poliwag [CP: 626] [IV: 13/15/15] Potential: 0.96
```

**Comment by LcTrKiD** [comment]
@matheussampaio do I need to use this exact config.json? Only the coordinates?

**Comment by matheussampaio** [comment]
@LcTrKiD This is my `config.json`, you're free to change whatever you want. Don't forget to comment here if you got a better experience/hour ratio. 😄

**Comment by nikhil-pandey** [comment]
```
[09:15:26] Encountered 1508 pokemon, 1431 caught, 0 released, 203 evolved, 1 never seen before
```
@matheussampaio 1431 caught 0 released? Pokemon storage isn't getting full?

**Comment by Nihisil** [comment]
@nikhil-pandey released stat is broken, it doesn't display real number.

**Comment by LcTrKiD** [comment]
@matheussampaio do i need to clone your fork?

**Comment by nikhil-pandey** [comment]
@LcTrKiD 
If you have already cloned this repo just do this.
```
git fetch origin pull/2491/head:more-is-better
git checkout more-is-better
```

**Comment by LcTrKiD** [comment]
Im getting this error after starting the bot: 
`Traceback (most recent call last):
  File "pokecli.py", line 452, in <module>
    main()
  File "pokecli.py", line 71, in main
    bot.tick()
  File "/Users/Shared/Pokecli/more-is-better/PokemonGo-Bot/pokemongo_bot/__init__.py", line 89, in tick
    if worker.work(tick_count = self.tick_count) == WorkerResult.RUNNING:
TypeError: work() got an unexpected keyword argument 'tick_count'`

**Comment by matheussampaio** [comment]
You're using a worker that I didn't add `def work(self, *args, **kwargs):`.

**Comment by w0rd3r** [comment]
Just tested with a Lucky Egg for 10 minutes or so, impressive: 

[18:56:08] Uptime : 0:10:26 | 1.24km walked | Level 25 (220,625 / 340,000, 64%) | +60,650 XP | 6,065 XP/m | Visited 5.00 /m | Caught 6.00 /m | Evolved 3.00 /m 


Thanks!

**Comment by nikhil-pandey** [comment]
Could you guys share your release config for pokemons

**Comment by LcTrKiD** [comment]
After 36 minutes using the bot I'm getting only this for more than 30 minutes:

[19:56:10] Waiting 4.44...
[19:56:17] Waiting 3.62...
[19:56:21] Waiting 4.45...
[19:56:26] Waiting 4.48...
[19:56:32] Waiting 4.33...
[19:56:37] Waiting 4.40...
[19:56:43] Waiting 4.05...
[19:56:48] Waiting 4.48...
[19:56:53] Waiting 4.41...
[19:56:58] Waiting 4.47...
[19:57:04] Uptime : 1:08:05 | 0.98km walked | Level 9 (15,775 / 17,000, 92%) | +43,175 XP | 634 XP/m | Visited 2.04 /m | Caught 1.18 /m | Evolved 0.03 /m 
[19:57:04] Waiting 4.12...
[19:57:10] Waiting 4.13...
[19:57:15] Waiting 4.31...

**Comment by matheussampaio** [comment]
@LcTrKiD Can you share your `config.json`?

**Comment by w0rd3r** [comment]
Uptime : 0:58:58 | 3.29km walked | Level 26 (292,835 / 390,000, 75%) | +216,080 XP | 3,662 XP/m | Visited 4.36 /m | Caught 3.85 /m | Evolved 1.61 /m 

After almost an hour and 2 lucky eggs

**Comment by matheussampaio** [comment]
Apparently, some thing around 1450 XP/m (87k XP/H) is what everyone is getting from this commit. If you have a lot of candies you can get a little boost evolving pokemons.

**Comment by LcTrKiD** [comment]
@matheussampaio this is my config.son

{
    "auth_service": "google",
    "username": "xxxxx@gmail.com",
    "password": "xxxxxxxxxxxx",
    "location": "36.830958, -2.4580580",
    "gmapkey": "xxxxxxxxxxx",
	"tasks": [
	        {
	            "type": "UpdateTitleStats",
	            "config": {
	                "min_interval": 60,
	                "stats": ["uptime", "km_walked", "level_stats", "xp_earned", "xp_per_minute", "stops_visited", "pokemon_caught", "pokemon_evolved"]
	            }
	        },
	        {
	            "type": "RecycleItems",
	            "config": {
	                "item_filter": {
	                    "Pokeball": {
	                        "keep": 80
	                    },
	                    "Greatball": {
	                        "keep": 80
	                    },
	                    "Ultraball": {
	                        "keep": 80
	                    },
	                    "Masterball": {
	                        "keep": 80
	                    },
	                    "Razz Berry": {
	                        "keep": 50
	                    },
	                    "Potion": {
	                        "keep": 0
	                    },
	                    "Super Potion": {
	                        "keep": 0
	                    },
	                    "Hyper Potion": {
	                        "keep": 0
	                    },
	                    "Max Potion": {
	                        "keep": 0
	                    },
	                    "Revive": {
	                        "keep": 0
	                    },
	                    "Max Revive": {
	                        "keep": 0
	                    }
	                }
	            }
	        },
	        {
	            "type": "CollectLevelUpReward"
	        },
	        {
	            "type": "IncubateEggs",
	            "config": {
	                "longer_eggs_first": true
	            }
	        },
	        {
	            "type": "CatchVisiblePokemon"
	        },
	        {
	            "type": "CatchLuredPokemon"
	        },
	        {
	            "type": "TransferPokemon"
	        },
	        {
	            "type": "SpinFort"
	        },
	        {
	            "type": "MoveToFort",
	            "config": {
	                "lure_attraction": true,
	                "lure_max_distance": 2000
	            }
	        }
	    ],
	    "map_object_cache_time": 5,
	    "max_steps": 150,
	    "forts": {
	        "avoid_circles": true,
	        "max_circle_size": 100
	    },
	    "websocket_server": true,
	    "walk": 80,
	    "action_wait_min": 0,
	    "action_wait_max": 1,
	    "debug": false,
	    "test": false,
	    "health_record": false,
	    "location_cache": false,
	    "distance_unit": "km",
	    "reconnecting_timeout": 15,
	    "evolve_captured": "Pidgey,Rattata,Zubat,Weedle,Drowzee,Spearow,Venonat,Caterpie,Krabby,Paras,Jigglypuff,Sandshrew,Vulpix,Meowth,Psyduck,Mankey,Tentacool,Ponyta,Slowpoke,Magnemite,Doduo,Seel,Voltorb,Horsea,Goldeen,Staryu,Ekans",
	    "catch_randomize_reticle_factor": 1.0,
	    "catch_randomize_spin_factor": 1.0,
	    "catch": {
	        "any": {
	            "catch_above_cp": 0,
	            "catch_above_iv": 0,
	            "logic": "or"
	        }
	    },
    "release": {
      "any": {"release_below_cp": 0, "release_below_iv": 0, "logic": "or"},
      "// Example of always releasing Rattata:": {},
      "Metapod": {"always_release": true},
	  "Kakuna": {"always_release": true},
	  "Pidgeotto": {"always_release": true},
	  "Raticate": {"always_release": true},
	  "Arbok": {"always_release": true},
	  "Sandslash": {"always_release": true},
	  "Golbat": {"always_release": true},
	  "// Example of keeping 2 or 10 stronger (based on CP) Pidgey:": {},
      "Pidgey": {"keep_best_cp": 10},
	  "Caterpie": {"keep_best_cp": 10},
	  "Weedle": {"keep_best_cp": 10},
	  "Rattata": {"keep_best_cp": 2},
	  "Ekans": {"keep_best_cp": 2},
	  "Zubat": {"keep_best_cp": 2},
	  "Magnemite": {"keep_best_cp": 2},
	  "Doduo": {"keep_best_cp": 2},
	  "Shellder": {"keep_best_cp": 2},
	  "Horsea": {"keep_best_cp": 2},
	  "Cubone": {"keep_best_cp": 2},
	  "Rhyhorn": {"keep_best_cp": 2},
	  "Paras": {"keep_best_cp": 2},
	  "Omanyte": {"keep_best_cp": 2},
	  "Jigglypuff": {"keep_best_cp": 2},
	  "Nidoran♂": {"keep_best_cp": 2},
	  "Venonat": {"keep_best_cp": 2},
	  "Spearow": {"keep_best_cp": 2},
	  "Sandshrew": {"keep_best_cp": 2},
	  "Meowth": {"keep_best_cp": 2},
	  "Psyduck": {"keep_best_cp": 2},
	  "Mankey": {"keep_best_cp": 2},
	  "Poliwag": {"keep_best_cp": 2},
	  "Staryu": {"keep_best_cp": 2},
	  "Abra": {"keep_best_cp": 2},
	  "Machop": {"keep_best_cp": 2},
	  "Tentacool": {"keep_best_cp": 2},
	  "Geodude": {"keep_best_cp": 2},
	  "Ponyta": {"keep_best_cp": 2},
	  "Drowzee": {"keep_best_cp": 2},
	  "Krabby": {"keep_best_cp": 2},
	  "Voltorb": {"keep_best_cp": 2},
	  "Cubone": {"keep_best_cp": 2},
	  "Goldeen": {"keep_best_cp": 2},
	  "Magikarp": {"keep_best_cp": 2},
	  "Bulbasur": {"keep_best_cp": 1, "keep_best_iv": 1},
	  "Charmander": {"keep_best_cp": 1, "keep_best_iv": 1},
	  "Squirtle": {"keep_best_cp": 1, "keep_best_iv": 1},
	  "Pikachu": {"keep_best_cp": 1, "keep_best_iv": 1},
	  "Growlithe": {"keep_best_cp": 1, "keep_best_iv": 1}, 
	  
      "// Example of keeping 2 stronger (based on IV) Zubat:": {},
      "// Zubat": {"keep_best_iv": 2},
      "// Also, it is working with any": {},
      "// any": {"keep_best_iv": 3},
      "// Example of keeping the 2 strongest (based on CP) and 3 best (based on IV) Zubat:": {},
      "// Zubat": {"keep_best_cp": 2, "keep_best_iv": 3}
    },
    "vips" : {
         "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {"catch_above_cp": 1200, "catch_above_iv": 0.9, "logic": "or" },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},

        "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "// Mew evolves to Mewtwo": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {},
		"Charizard": {}

    }
}

**Comment by bibibkamal** [comment]
I'm sorry I'm new but how come my terminal didn't show its stats?

**Comment by CoderFX** [comment]
I get this error:

> Traceback (most recent call last):
  File "pokecli.py", line 452, in <module>
    main()
  File "pokecli.py", line 71, in main
    bot.tick()
  File "G:\Pokemon\PokemonGo-Bot - ptc\pokemongo_bot\__init__.py", line 89, in tick
    if worker.work(tick_count = self.tick_count) == WorkerResult.RUNNING:
  File "G:\Pokemon\PokemonGo-Bot - ptc\pokemongo_bot\cell_workers\recycle_items.py", line 25, in work
    item_count_dict = self.bot.item_inventory_count('all')
  File "G:\Pokemon\PokemonGo-Bot - ptc\pokemongo_bot\__init__.py", line 429, in item_inventory_count
    'GET_INVENTORY']['inventory_delta']['inventory_items']
KeyError: 'inventory_delta'

**Comment by bibibkamal** [comment]
File "pokecli.py", line 220, in <module>
    main()
  File "pokecli.py", line 211, in main
    bot.take_step()
  File "/Users/bibibkamal/Desktop/PokemonGO/PokemonGo-Bot/pokemongo_bot/__init__.py", line 36, in take_step
    self.stepper.take_step()
  File "/Users/bibibkamal/Desktop/PokemonGO/PokemonGo-Bot/pokemongo_bot/stepper.py", line 62, in take_step
    self._work_at_position(position[0], position[1], position[2], True)
  File "/Users/bibibkamal/Desktop/PokemonGO/PokemonGo-Bot/pokemongo_bot/stepper.py", line 139, in _work_at_position
    self.bot.work_on_cell(cell, position, pokemon_only)
  File "/Users/bibibkamal/Desktop/PokemonGO/PokemonGo-Bot/pokemongo_bot/__init__.py", line 96, in work_on_cell
    hack_chain = worker.work()
  File "/Users/bibibkamal/Desktop/PokemonGO/PokemonGo-Bot/pokemongo_bot/cell_workers/seen_fort_worker.py", line 110, in work
    raise RuntimeError(message)
untimeError: Stopped at Pokestop and did not find experience, items or information about the stop cooldown. You are probably softbanned. Try to play on your phone, if pokemons always ran away and you find nothing in PokeStops you are indeed softbanned. Please try again in a few hours.

i got this error every 10 to 12 minutes. Have to restart my bot again, I'm not soft banned tho. Any help?

**Comment by theVDude** [comment]
Uptime : 0:31:21 | 2.36km walked | Level 29 (562,835 / 650,000, 86%) | +126,160 XP | 4,069 XP/m | Visited 4.68 /m | Caught 2.97 /m | Evolved 2.39 /m

That's my half hour with a lucky egg. Feels pretty good, man!

**Comment by CoderFX** [comment]
Can you post full instructions on how to make this work? Getting softbanned too often

**Comment by matheussampaio** [comment]
@CoderFX WIP

**Comment by LcTrKiD** [comment]
@matheussampaio I have a lot of pokestops at this coordinates but still getting this:

[22:28:39] Login to Pokemon Go successful. 
[22:28:40] 
[22:28:40] --- LcTrKiD --- 
[22:28:40] Level: 26 (Next Level: 88524 XP) (Total: 1011476 XP) 
[22:28:40] Pokemon Captured: 3152 | Pokestops Visited: 2392 
[22:28:40] Pokemon Bag: 179/250 
[22:28:40] Items: 288/350 
[22:28:40] Stardust: 79503 | Pokecoins: 0 
[22:28:40] PokeBalls: 100 | GreatBalls: 21 | UltraBalls: 10 
[22:28:40] RazzBerries: 21 | BlukBerries: 0 | NanabBerries: 0 
[22:28:40] LuckyEgg: 0 | Incubator: 0 | TroyDisk: 6 
[22:28:40] Potion: 0 | SuperPotion: 30 | HyperPotion: 30 
[22:28:40] Incense: 8 | IncenseSpicy: 0 | IncenseCool: 0 
[22:28:40] Revive: 30 | MaxRevive: 0 
[22:28:40] 
[22:28:40] 
[22:28:41] Starting PokemonGo Bot.... 
[22:28:41] Uptime : 0:00:05 | 0.00km walked | Level 26 (301,476 / 390,000, 77%) | +0 XP | 0 XP/m | Visited 0.00 /m | Caught 0.00 /m | Evolved 0.00 /m 
[22:28:42] Current egg hatches in 4.42 km
[22:28:43] Waiting 2.89...
[22:28:46] Waiting 5.00...
[22:28:52] Waiting 5.00...
[22:28:57] Waiting 5.00...
[22:29:02] Waiting 5.00...
[22:29:08] Waiting 4.45...
[22:29:13] Waiting 5.00...
[22:29:18] Waiting 5.00...
[22:29:24] Waiting 5.00...
[22:29:29] Waiting 5.00...
[22:29:35] Waiting 4.40...
[22:29:40] Waiting 5.00...

**Comment by Erody** [comment]
Getting the Waiting spam as well now. Was running just fine for a few hours before that. Even after restarting it just spams Waiting ...

**Comment by matheussampaio** [comment]
@LcTrKiD Everyone is getting the same, server problem or something changed. Please, use slack for troubleshooting.

**Comment by Segflow** [comment]
`21:36:51] Starting PokemonGo Bot....
[21:36:53] Uptime : 0:00:10 | 0.00km walked | Level 28 (317,970 / 550,000, 57%) | +0 XP | 0 XP/m | Visited 0.00 /m | Caught 0.00 /m | Evolved 0.00 /m
[21:36:59] Current egg hatches in 5.00 km
[21:37:02] Waiting 5.00...
[21:37:09] Waiting 5.00...
[21:37:14] Waiting 5.00...
[21:37:20] Waiting 5.00...
[21:37:26] Waiting 4.03...
[21:37:31] Waiting 5.00...
[21:37:36] Waiting 5.00...
[21:37:42] Waiting 5.00...
[21:37:47] Waiting 5.00...`

same config as yours(location= central park), can you help please?

File: pokemongo_bot/cell_workers/catch_lured_pokemon.py [hunk 1]
```diff
@@ -1,36 +1,47 @@
...
+
+            if encounter_id:
+                self.catch_pokemon({
+                    'encounter_id': encounter_id,
+                    'fort_id': fort['id'],
+                    'latitude': fort['latitude'],
+                    'longitude': fort['longitude']
+                })
+
+            time.sleep(1)
```
Comment (mako-taco) [inline pokemongo_bot/cell_workers/catch_lured_pokemon.py]: can we use `config.min_time_between_actions` or whatever the value was?

File: pokemongo_bot/cell_workers/handle_soft_ban.py [hunk 2]
```diff
@@ -9,7 +9,10 @@
 
 
 class HandleSoftBan(BaseTask):
-    def work(self):
+    def work(self, *args, **kwargs):
```
Comment (mako-taco) [inline pokemongo_bot/cell_workers/handle_soft_ban.py]: this worker runs in less than 50ms when `should_run()` returns false, this is probably unneeded

File: pokemongo_bot/cell_workers/incubate_eggs.py [hunk 3]
```diff
@@ -19,7 +19,10 @@ def initialize(self):
     def _process_config(self):
         self.longer_eggs_first = self.config.get("longer_eggs_first", True)
 
-    def work(self):
+    def work(self, *args, **kwargs):
+        if kwargs.get('tick_count', -1) % 1:
```
Comment (mako-taco) [inline pokemongo_bot/cell_workers/incubate_eggs.py]: `x % 1` is always 0, no?

File: pokemongo_bot/cell_workers/catch_lured_pokemon.py (hunk 1 above)
Comment (matheussampaio) [inline pokemongo_bot/cell_workers/catch_lured_pokemon.py]: Yes, when we got the bot working again, we'll do it right and remove the [XGH](https://gist.github.com/banaslee/4147370) code. 😄

File: pokemongo_bot/cell_workers/incubate_eggs.py (hunk 3 above)
Comment (matheussampaio) [inline pokemongo_bot/cell_workers/incubate_eggs.py]: Yes, I tried to write 10.

**Comment by w0rd3r** [comment]
Hello, does this still work with the new API?

**Comment by douglascamata** [comment]
@matheussampaio we're back up again, please your PR! :D

**Comment by jokoast** [comment]
Update it please 😁 I need a strong xp/hour merge :D

**Comment by matheussampaio** [comment]
I'm busy today, but tomorrow I'll update.

**Comment by jokoast** [comment]
Thx dude ;). Your work is really welcome.

**Comment by icecubeice** [comment]
Hey somehow I cant pass source bin/activate:
$ source bin/activate
bash: bin/activate: No such file or directory

**Comment by navdeepz** [comment]
I am using the same location (40.774626,-73.972706) but always get "can't find a nearby fort to walk to" I even used location specified in reddit post, still no luck. https://www.reddit.com/r/pokemongo/comments/4thygt/central_park_nyc_is_the_best_farm_spot_these_4/

[14:09:31] can't find a nearby fort to walk to
[14:09:31] Waiting 4.57...
[14:09:36] can't find a nearby fort to walk to
[14:09:36] Waiting 5.00...
[14:09:41] can't find a nearby fort to walk to
[14:09:41] Waiting 5.00...
[14:09:53] can't find a nearby fort to walk to
[14:09:53] Waiting 4.50...
[14:09:58] can't find a nearby fort to walk to

**Comment by cmezh** [comment]
I think, currently it is not adapted to last API changes.

**Comment by navdeepz** [comment]
Did you mean the code on my system is not adapted or this pull is not adapted to the latest API change?

**Comment by cmezh** [comment]
This pull is not adapted. Last commit was on 4th Aug, but new API was released on 7th Aug after Niantic's surprise.

**Comment by matheussampaio** [comment]
Deprecated. Check #2889.

**Comment by jomarocas** [comment]
```
    "tasks": [
      {
        "type": "HandleSoftBan"
      },
      {
        "type": "CollectLevelUpReward"
      },
      {
        "type": "IncubateEggs",
        "config": {
          "longer_eggs_first": true
        }
      },
      {
        "type": "TransferPokemon"
      },
      {
        "type": "EvolvePokemon",
        "config": {
            "evolve_all": "none",
            "first_evolve_by": "cp",
            "evolve_above_cp": 500,
            "evolve_above_iv": 0.8,
            "logic": "or",
            "evolve_speed": 20,
            "use_lucky_egg": false
        }
      },
      {
        "type": "RecycleItems",
        "config": {
          "item_filter": {
            "Pokeball":       { "keep" : 50 },
            "Ultraball":      { "keep" : 80},
            "Potion":         { "keep" : 0 },
            "Super Potion":   { "keep" : 0 },
            "Hyper Potion":   { "keep" : 20 },
            "Revive":         { "keep" : 20 },
            "Razz Berry":     { "keep" : 60 }
          }
        }
      },
      {
        "type": "CatchVisiblePokemon"
      },
      {
        "type": "CatchLuredPokemon"
      },
      {
        "type": "SpinFort"
      },
      {
        "type": "MoveToFort",
        "config": {
            "lure_attraction": true,
            "lure_max_distance": 2000
        }
      },
      {
        "type": "FollowSpiral",
        "config": {
          "diameter": 4,
          "step_size": 70
        }
      }
    ],
    "map_object_cache_time": 5,
    "forts": {
      "avoid_circles": true,
      "max_circle_size": 50
    },
    "websocket_server": false,
    "walk": 19.16,
    "action_wait_min": 1,
    "action_wait_max": 4,
    "debug": false,
    "test": false,
    "health_record": true,
    "location_cache": true,
    "distance_unit": "km",
    "reconnecting_timeout": 15,
    "evolve_captured": "NONE",
    "catch_randomize_reticle_factor": 1.0,
    "catch_randomize_spin_factor": 1.0,
    "catch": {
      "any": {"catch_above_cp": 0, "catch_above_iv": 0, "logic": "or"},
      "// Example of always catching Rattata:": {},
      "// Rattata": { "always_catch" : true }
    },
    "release": {
      "any": {"release_below_cp": 0, "release_below_iv": 0, "logic": "or"},
      "Pidgey": {"keep_best_cp": 10},
      "Caterpie": {"keep_best_cp": 2},
      "Weedle": {"keep_best_cp": 2},
      "Rattata": {"keep_best_cp": 2},
      "Ekans": {"keep_best_cp": 2},
      "Zubat": {"keep_best_cp": 2},
      "Magnemite": {"keep_best_cp": 2},
      "Doduo": {"keep_best_cp": 2},
      "Shellder": {"keep_best_cp": 2},
      "Horsea": {"keep_best_cp": 2},
      "Cubone": {"keep_best_cp": 2},
      "Rhyhorn": {"keep_best_cp": 2},
      "Paras": {"keep_best_cp": 2},
      "Omanyte": {"keep_best_cp": 2},
      "Jigglypuff": {"keep_best_cp": 2},
      "Nidoran♂": {"keep_best_cp": 2},
      "Venonat": {"keep_best_cp": 2},
      "Spearow": {"keep_best_cp": 2},
      "Sandshrew": {"keep_best_cp": 2},
      "Meowth": {"keep_best_cp": 2},
      "Psyduck": {"keep_best_cp": 2},
      "Mankey": {"keep_best_cp": 2},
      "Poliwag": {"keep_best_cp": 2},
      "Staryu": {"keep_best_cp": 2},
      "Abra": {"keep_best_cp": 2},
      "Machop": {"keep_best_cp": 2},
      "Tentacool": {"keep_best_cp": 2},
      "Geodude": {"keep_best_cp": 2},
      "Ponyta": {"keep_best_cp": 2},
      "Drowzee": {"keep_best_cp": 2},
      "Krabby": {"keep_best_cp": 2},
      "Voltorb": {"keep_best_cp": 2},
      "Cubone": {"keep_best_cp": 2},
      "Goldeen": {"keep_best_cp": 2},
      "Magikarp": {"keep_best_cp": 2},
      "Bulbasur": {"keep_best_cp": 2, "keep_best_iv": 1},
      "Charmander": {"keep_best_cp": 2, "keep_best_iv": 1},
      "Squirtle": {"keep_best_cp": 2, "keep_best_iv": 1},
      "Pikachu": {"keep_best_cp": 2, "keep_best_iv": 1},
      "Growlithe": {"keep_best_cp": 2, "keep_best_iv": 1}
    },
    "vips" : {
         "Any pokemon put here directly force to use Berry & Best Ball to capture, to secure the capture rate!": {},
        "any": {"catch_above_cp": 1200, "catch_above_iv": 0.9, "logic": "or" },
        "Lapras": {},
        "Moltres": {},
        "Zapdos": {},
        "Articuno": {},

        "// S-Tier pokemons (if pokemon can be evolved into tier, list the representative)": {},
        "Mewtwo": {},
        "Dragonite": {},
        "Snorlax": {},
        "// Mew evolves to Mewtwo": {},
        "Mew": {},
        "Arcanine": {},
        "Vaporeon": {},
        "Gyarados": {},
        "Exeggutor": {},
        "Muk": {},
        "Weezing": {},
        "Flareon": {}

    }
```

**Pull request closed without merging**